### PR TITLE
Feature : Support for adding external camera

### PR DESCRIPTION
**Pull request by maddygoround**
[sc-17216]

### CODE REVIEW

File: src/core/sources/WebRTC.ts [hunk 1]
```diff
@@ -103,38 +135,37 @@ export const RoomParticipant = {
...
-              addSource(source)
-            } else {
-              addSource(source)
-            }
+          const { id, participantId, type, isExternal } = room.getTrack(x.id)
+          if (type === 'camera') {
+            const { displayName, meta } = room.getParticipant(participantId)
+
+            const srcObject = new MediaStream([])
+            participantStreams[id] = srcObject
```
Comment (mikekanter) [inline src/core/sources/WebRTC.ts:144]: Why is `srcObject` simply a `new MediaStream`? why not use the `mediaStreamTrack` from `room.getTrack`

Comment (maddygoround) [reply src/core/sources/WebRTC.ts:144]: The final result used by weebrtc transformer is an srcObject. We update the srcobject with  mediastreamtrack when calling updateParticipant.

File: src/helpers/sceneless-project.ts [hunk 2]
```diff
@@ -1776,6 +1795,37 @@ export const commands = (_project: ScenelessProject) => {
...
+    async addParticipantTrack(
+      trackId: string,
+      props: Partial<ParticipantProps> = {
+        isMuted: true,
+        isHidden: false,
+        volume: 0,
+      },
+      type: ParticipantType = 'camera',
+    ) {
+      return await commands.addParticipant(trackId, props, type)
```
Comment (mikekanter) [inline src/helpers/sceneless-project.ts:1808]: Down in line 1829, we have:
```typescript
 async addParticipant(
      participantId: string,
      props: Partial<ParticipantProps> = {},
      type: ParticipantType = 'camera',
)
```

But here, we call `addParticipant`, and pass `trackId` in as the `participantId`? This is confusing to me.

Comment (maddygoround) [reply src/helpers/sceneless-project.ts:1808]: Add a track to the stream in the same way you would add a participant.
The primary camera is assigned to the participant, while all other external cameras are assigned to ParticipantTrack.

We don't actually need a separate function, but it's there for the sake of differentiation for end customers. So they are aware of two functions, addParticipant and addParticipantTrack, both of which perform the same function.

**Review by mikekanter — APPROVED** [review]
alright; approving

**Pull request closed without merging**